common/client-core/src/client/base_client/mod.rs

```diff
@@ -263,6 +272,16 @@ where
         self
     }
 
+    /// Set network details for domain fronting support.
+    ///
+    /// When provided, the client will use network details (which include front_hosts)
+    /// to construct HTTP clients with domain fronting enabled.
+    #[must_use]
+    pub fn with_network_details(mut self, network_details: NymNetworkDetails) -> Self {
```

Comment:
if it's just for domain fronting, wouldn't it make more sense to only pass `Vec<ApiUrl>`?
this is for few reasons:
1. passing whole `NymNetworkDetails` might give wrong impression as if this allowed you to control, for example, contract endpoints, which is not the intention here
2. `NymNetworkDetails` allows for `None` value for the api urls, which again, is not what we want here
3. it is a more explicit way of stating we meant to only change domain fronting